Plutarch/Internal.hs

```diff
@@ -60,6 +60,20 @@ hashRawTerm' (RHoisted (HoistedTerm hash _)) = flip hashUpdate ("8" :: BS.ByteSt
 hashRawTerm :: RawTerm -> Dig
 hashRawTerm t = hashFinalize . hashRawTerm' t $ hashInit
 
+data TermResult = TermResult
+  { getTerm :: RawTerm
+  , getDeps :: [HoistedTerm]
+  }
+
+mapTerm :: (RawTerm -> RawTerm) -> TermResult -> TermResult
+mapTerm f (TermResult t d) = TermResult (f t) d
+
+mergeTerms :: (RawTerm -> RawTerm -> RawTerm) -> TermResult -> TermResult -> TermResult
+mergeTerms f x y = TermResult (f (getTerm x) (getTerm y)) (getDeps x <> getDeps y)
+
+mkNew :: RawTerm -> TermResult
+mkNew r = TermResult r []
```

Comment:
Name this `mkTermRes` maybe? I understand that this isn't exported so it hardly matters but still.

Reply:
Thanks.